app/src/ui/history/compare.tsx

```diff
@@ -527,9 +533,24 @@ export class CompareSidebar extends React.Component<
     this.textbox = textbox
   }
 
-  private onNotificationBannerDismissed = () => {
+  private onNotificationBannerDismissed = (reason: DismissalReason) => {
     this.props.dispatcher.setDivergingBranchBannerVisibility(false)
     this.props.dispatcher.recordDivergingBranchBannerDismissal()
+
+    if (reason === 'close') {
+      this.setState({ hasConsumedNotification: false })
+      return
```

Comment:
This seems like a good job for a switch statement, particularly if you don't care about `'merge'`:

```ts

    switch (reason) {
      case 'close':
        this.setState({ hasConsumedNotification: false })
        return
      case 'compare':
        this.setState({ hasConsumedNotification: true })
        return
      case 'merge':
      default:
        return
    }
```

Reply:
I don't care about merge yet 😉. But, 👍 on the switch.